src/Http/Http.Abstractions/src/Extensions/HttpResponseWritingExtensions.cs

```diff
@@ -61,7 +61,7 @@ public static Task WriteAsync(this HttpResponse response, string text, Encoding
             }
 
             byte[] data = encoding.GetBytes(text);
-            return response.Body.WriteAsync(data, 0, data.Length, cancellationToken);
+            return response.BodyPipe.WriteAsync(new Memory<byte>(data, 0, data.Length), cancellationToken).AsTask();
```

Comment:
A couple of things:
- This ends up creating a temporary buffer
- It allocates a new task

We need to do a few things here:
- Use GetMemory() and write directly into it
- Don't use `AsTask()` unless the ValueTask returned from FlushAsync Isn't completed.
  - Actually, make a local function that does the await, instead of using AsTask.

Reply:
If I use GetMemory, I would need to `await StartAsync`.

Reply:
You would need to call StartAsync, see if it is completed, if it isn't then await it.

Reply:
@davidfowl for future reference, why did this create a temporary buffer before?